import{_ as e,c as a,o as r,az as l}from"./chunks/framework.DiV9SzYZ.js";const h=JSON.parse('{"title":"Bioconductor (R)","description":"","frontmatter":{},"headers":[],"relativePath":"comparisons/bioconductor.md","filePath":"comparisons/bioconductor.md","lastUpdated":null}'),i={name:"comparisons/bioconductor.md"};function o(n,t,s,d,g,c){return r(),a("div",null,t[0]||(t[0]=[l('<h1 id="bioconductor" tabindex="-1">Bioconductor (R) <a class="header-anchor" href="#bioconductor" aria-label="Permalink to &quot;Bioconductor (R) {#bioconductor}&quot;">â€‹</a></h1><ul><li><p><a href="https://www.bioconductor.org/" target="_blank" rel="noreferrer">Bioconductor</a> represents an ecosystem of related genomics tools written in the R programming language</p></li><li><p>Tools written in other languages, such as C and Python, are also included with R wrappers available</p></li><li><p>Launched over two decades ago, it now hosts over 2000 packages for bioinformatics and related fields</p></li><li><p>Packages cover a broard range of methods for the analysis and manipulation of genomic data</p></li></ul><p><strong>A few equivalent applications between BioJulia and Bioconductor:</strong></p><table tabindex="0"><thead><tr><th style="text-align:left;">Application</th><th style="text-align:left;">BioJulia packages</th><th style="text-align:left;">Bioconductor packages</th></tr></thead><tbody><tr><td style="text-align:left;">Data structures</td><td style="text-align:left;"><a href="https://biojulia.dev/BioSymbols.jl/stable/" target="_blank" rel="noreferrer">BioSymbols</a>, <a href="https://github.com/BioJulia/KmerAnalysis.jl" target="_blank" rel="noreferrer">KmerAnalysis</a>, <a href="https://biojulia.dev/IntervalTrees.jl/stable/" target="_blank" rel="noreferrer">IntervalTrees</a></td><td style="text-align:left;"><code>Biobase</code></td></tr><tr><td style="text-align:left;">Sequence annotation</td><td style="text-align:left;"><a href="https://biojulia.dev/GenomicFeatures.jl/stable/" target="_blank" rel="noreferrer">GenomicFeatures</a>, <a href="https://biojulia.dev/GenomicAnnotations.jl/dev/" target="_blank" rel="noreferrer">GenomicAnnotations</a>, <a href="https://github.com/BioJulia/FormatSpecimens.jl" target="_blank" rel="noreferrer">FormatSpecimens</a></td><td style="text-align:left;"><code>AnnotationHub</code></td></tr><tr><td style="text-align:left;">Input/Output</td><td style="text-align:left;"><a href="https://biojulia.dev/FASTX.jl/stable/" target="_blank" rel="noreferrer">FASTX</a>, <a href="https://biojulia.dev/XAM.jl/stable/" target="_blank" rel="noreferrer">XAM</a>, <a href="https://biojulia.dev/BigWig.jl/dev/" target="_blank" rel="noreferrer">BigWig</a>,...</td><td style="text-align:left;"><code>BiocIO</code>, <code>Biostrings</code>, <code>ShortRead</code>,...</td></tr><tr><td style="text-align:left;">Sequence alignment</td><td style="text-align:left;"><a href="https://biojulia.dev/BioSequences.jl/stable/" target="_blank" rel="noreferrer">BioSequences</a>, <a href="https://biojulia.dev/BioAlignments.jl/stable/" target="_blank" rel="noreferrer">BioAlignments</a></td><td style="text-align:left;"><code>Biostrings</code></td></tr><tr><td style="text-align:left;">Expression analysis</td><td style="text-align:left;"><a href="https://biojulia.dev/SingleCellProjections.jl/dev/" target="_blank" rel="noreferrer">SingleCellProjections</a></td><td style="text-align:left;"><code>DESeq2</code></td></tr></tbody></table><p><strong>A few package/ecosystem equivalents between Julia and R:</strong></p><table tabindex="0"><thead><tr><th style="text-align:left;">Application</th><th style="text-align:left;">Julia</th><th style="text-align:left;">R</th></tr></thead><tbody><tr><td style="text-align:left;">Data manipulation/analysis</td><td style="text-align:left;"><a href="https://dataframes.juliadata.org/stable/" target="_blank" rel="noreferrer">DataFrames</a>, <a href="https://csv.juliadata.org/stable/" target="_blank" rel="noreferrer">CSV</a>, <a href="https://www.queryverse.org/Query.jl/stable/" target="_blank" rel="noreferrer">Query</a>, <a href="https://tidierorg.github.io/Tidier.jl/dev/" target="_blank" rel="noreferrer">Tidier</a></td><td style="text-align:left;"><code>tibble</code>, <code>dplyr</code>, <code>tidyverse</code></td></tr><tr><td style="text-align:left;">Plotting/visualization</td><td style="text-align:left;"><a href="http://gadflyjl.org/stable/" target="_blank" rel="noreferrer">Gadfly</a>, <a href="https://www.queryverse.org/VegaLite.jl/stable/" target="_blank" rel="noreferrer">VegaLite</a>, <a href="https://docs.makie.org/stable/" target="_blank" rel="noreferrer">Makie</a>, <a href="https://github.com/TidierOrg/TidierPlots.jl" target="_blank" rel="noreferrer">TidierPlots</a></td><td style="text-align:left;"><code>ggplot2</code>, <code>vegalite</code>, <code>plotly</code></td></tr><tr><td style="text-align:left;">Statistical analysis</td><td style="text-align:left;"><a href="https://docs.julialang.org/en/v1/stdlib/Statistics/" target="_blank" rel="noreferrer">Statistics</a>, <a href="https://github.com/JuliaStats/HypothesisTests.jl" target="_blank" rel="noreferrer">HypothesisTests</a>, <a href="https://github.com/JuliaStats/GLM.jl" target="_blank" rel="noreferrer">GLM</a></td><td style="text-align:left;"><code>stats</code></td></tr><tr><td style="text-align:left;">Machine learning</td><td style="text-align:left;"><a href="https://fluxml.ai/Flux.jl/stable/" target="_blank" rel="noreferrer">Flux</a>, <a href="https://sciml.ai/" target="_blank" rel="noreferrer">SciML</a>, <a href="https://alan-turing-institute.github.io/MLJ.jl/stable/" target="_blank" rel="noreferrer">MLJ</a>, <a href="https://fluxml.ai/Zygote.jl/stable/" target="_blank" rel="noreferrer">Zygote</a></td><td style="text-align:left;"><code>mlr3</code>, <code>caret</code>, <code>tidymodels</code>, <code>Deriv</code></td></tr><tr><td style="text-align:left;">Numerical mathematics</td><td style="text-align:left;"><a href="https://docs.julialang.org/en/v1/stdlib/LinearAlgebra/" target="_blank" rel="noreferrer">LinearAlgebra</a>, <a href="https://iterativesolvers.julialinearalgebra.org/dev/" target="_blank" rel="noreferrer">IterativeSolvers</a></td><td style="text-align:left;"><code>Matrix</code>, <code>pracma</code>, <code>deSolve</code></td></tr><tr><td style="text-align:left;">Web applications</td><td style="text-align:left;"><a href="https://genieframework.com/" target="_blank" rel="noreferrer">Genie</a>, <a href="https://franklinjl.org/" target="_blank" rel="noreferrer">Franklin</a></td><td style="text-align:left;"><code>shiny</code></td></tr></tbody></table><p><strong>A few notable differences between Julia and R:</strong></p><table tabindex="0"><thead><tr><th style="text-align:left;">Julia</th><th style="text-align:left;">R</th></tr></thead><tbody><tr><td style="text-align:left;">High-level, general-purpose compiled language</td><td style="text-align:left;">High-level, interpreted language for statistical computing</td></tr><tr><td style="text-align:left;">Dynamically typed with multiple dispatch and optional type annotations</td><td style="text-align:left;">Dynamically typed without type annotations support</td></tr><tr><td style="text-align:left;">Built-in parallelism via threads, coroutines (Tasks)</td><td style="text-align:left;">Parallelism via external libraries (BLAS, <code>parallel</code>,...)</td></tr><tr><td style="text-align:left;">Extensive metaprogramming (Lisp-like macros, generated functions,...)</td><td style="text-align:left;">Less extensive metaprogramming (function factories, expression manipulation,...)</td></tr><tr><td style="text-align:left;">Single implementation available (JuliaLang)</td><td style="text-align:left;">Multiple implementations available (pqR, Renjin,...)</td></tr></tbody></table><p><strong>To transition from R to Julia:</strong></p><ul><li><p>See <a href="https://docs.julialang.org/en/v1/manual/noteworthy-differences/#Noteworthy-differences-from-R" target="_blank" rel="noreferrer">Noteworthy differences from R</a> in the Julia manual for a more in-depth comparison</p></li><li><p>Use <a href="https://github.com/JuliaInterop/RCall.jl.git" target="_blank" rel="noreferrer">RCall.jl</a> to seamlessly integrate R code into your Julia project</p></li><li><p>Use <a href="https://cran.r-project.org/web/packages/JuliaCall/readme/README.html" target="_blank" rel="noreferrer">JuliaCall</a> to seamlessly integrate Julia code into your R project</p></li><li><p>Check out the <a href="https://tidierorg.github.io/Tidier.jl/dev/" target="_blank" rel="noreferrer">Tidier.jl ecosystem</a> for packages similar to those commonly found in <code>tidyverse</code></p></li></ul>',10)]))}const p=e(i,[["render",o]]);export{h as __pageData,p as default};
